### PR TITLE
Made Timeframes Editable for Single Charts

### PR DESCRIPTION
**Pull request by chakrabortyr**
## Description
Prior to this change, single charts were not editable via the "Edit Timeframe for Selected Charts" button on the top of the Report Generator tab. This change remedies that. To not have 3 pieces of functionality that do the exact same thing, you can no longer edit timeframes by clicking the date range in the chart table.

BEFORE:
screenshot from 2017-09-20 11-29-44

AFTER:
screenshot from 2017-09-20 11-30-05

## Motivation and Context
The prior behavior was counterintuitive.

## Tests performed
Naturally

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project as found in the **CONTRIBUTING** document.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: html/gui/js/report_builder/ChartDateEditor.js [hunk 1]
```diff
@@ -244,22 +244,16 @@ XDMoD.Reporting.ChartDateEditor = Ext.extend(Ext.Window,  {
...
         }
 
-        if (chart_config.type.toLowerCase() == 'user defined') {
-
-           rdoGrpTimeframeMode.items.get(0).setValue(true);
-
-           this.start_date_field.setValue(chart_config.start);
-           this.end_date_field.setValue(chart_config.end);
-
+        if (chart_config.type && chart_config.type.toLowerCase() === 'user defined') {
```
Comment (plessbd) [inline html/gui/js/report_builder/ChartDateEditor.js]: why did this code block change?  does not seem related to your change...

File: html/gui/js/report_builder/ReportCreatorGrid.js [hunk 2]
```diff
@@ -491,7 +491,7 @@ XDMoD.ReportCreatorGrid = Ext.extend(Ext.Panel,  {
 
             var getData = sm.getSelections();
 
-            if (getData.length >= 2) {
```
Comment (plessbd) [inline html/gui/js/report_builder/ReportCreatorGrid.js]: is there a reason this was forced to be more than two to start with?

File: html/gui/js/report_builder/ChartDateEditor.js (hunk 1 above)
Comment (chakrabortyr) [reply html/gui/js/report_builder/ChartDateEditor.js]: While debugging my own code, I noticed an error when chart_config.type was null, so I figured I'd take care of it as well.

File: html/gui/js/report_builder/ReportCreatorGrid.js [hunk 3]
```diff
@@ -1,3 +1,5 @@
+/* eslint-disable */
```
Comment (plessbd) [inline html/gui/js/report_builder/ReportCreatorGrid.js]: We cant just disable the linting on an entire file.

File: html/gui/js/report_builder/Reporting.js [hunk 4]
```diff
@@ -1,3 +1,5 @@
+/* eslint-disable */
```
Comment (plessbd) [inline html/gui/js/report_builder/Reporting.js]: we cant just disable the linting on an entire file.

File: html/gui/js/report_builder/ChartDateEditor.js [hunk 5]
```diff
@@ -244,22 +244,16 @@ XDMoD.Reporting.ChartDateEditor = Ext.extend(Ext.Window,  {
...
 
-        if (chart_config.type.toLowerCase() == 'user defined') {
-
-           rdoGrpTimeframeMode.items.get(0).setValue(true);
-
-           this.start_date_field.setValue(chart_config.start);
-           this.end_date_field.setValue(chart_config.end);
-
+        /* eslint-disable */
+        if (chart_config.type && chart_config.type.toLowerCase() === 'user defined') {
```
Comment (plessbd) [inline html/gui/js/report_builder/ChartDateEditor.js]: if the only changes are adding the `===` and checking the existance og `chart_config_type` this should not need linting disabled.

**Comment by plessbd** [comment]
I think the rule changes should be limited in scope as much as possible instead of making these global, make them for the specific file (https://eslint.org/docs/user-guide/configuring#configuring-rules) with comments of why there were set the way they are, so that in the future if we want to try to fix them we can.

File: html/gui/js/report_builder/ChartDateEditor.js [hunk 6]
```diff
@@ -244,22 +244,17 @@ XDMoD.Reporting.ChartDateEditor = Ext.extend(Ext.Window,  {
...
         }
 
-        if (chart_config.type.toLowerCase() == 'user defined') {
-
-           rdoGrpTimeframeMode.items.get(0).setValue(true);
-
-           this.start_date_field.setValue(chart_config.start);
-           this.end_date_field.setValue(chart_config.end);
-
+        /* eslint-disable */
```
Comment (jpwhite4) [inline html/gui/js/report_builder/ChartDateEditor.js]: Please don't disable the linter. If there is a rule that you don't like then we can discuss and remove it from the project.

File: html/gui/js/report_builder/Reporting.js [hunk 7]
```diff
@@ -1,3 +1,5 @@
+/* eslint-disable indent*/
```
Comment (plessbd) [inline html/gui/js/report_builder/Reporting.js]: why the different styles of this disable
`/* eslint-disable indent*/`
vs
`/* eslint-disable indent: 0`
I think we should try to be consistent
